core/src/main/scala/org/apache/spark/shuffle/BlockStoreShuffleReader.scala

```diff
@@ -41,6 +43,20 @@ private[spark] class BlockStoreShuffleReader[K, C](
 
   private val dep = handle.dependency
 
+  private def fetchContinuousBlocksInBatch: Boolean = {
+    val conf = SparkEnv.get.conf
+    val compressed = conf.get(config.SHUFFLE_COMPRESS)
+    val featureEnabled = conf.get(config.SHUFFLE_FETCH_CONTINUOUS_BLOCKS_IN_BATCH)
+    val serializerRelocatable = dep.serializer.supportsRelocationOfSerializedObjects
+    // The batch fetching feature only works for reading consolidate file written by
+    // SortShuffleWriter or UnsafeShuffleWriter.
+    val readConsolidateFile = !handle.isInstanceOf[BypassMergeSortShuffleHandle[_, _]]
+
+    readConsolidateFile && featureEnabled && endPartition - startPartition > 1 &&
```

Comment:
How about we add a log here when featureEnabled is true while the final decision is false to indicate users that fetchContinuous is not actually used due to incompatible configures of serializer or compression codec.

Reply:
Make sense, done in d9ea5af.